homeassistant/components/netgear/__init__.py

```diff
@@ -92,9 +86,10 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Unload a config entry."""
     unload_ok = await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
 
-    # stop the coordinator
-    remove_dummy_update = hass.data[DOMAIN][entry.unique_id][KEY_STOP_COORDINATOR]
-    remove_dummy_update()
+    # remove new device listeners from the coordinator
+    new_device_listeners = hass.data[DOMAIN][entry.unique_id][KEY_NEW_DEVICE_LISTENERS]
+    for new_device_listener in new_device_listeners:
+        new_device_listener()
```

Comment:
Please use `entry.async_on_unload` instead when setting up the listener.

Reply:
thanks, done